### PR TITLE
breadth-first search for dependents upgrade

### PR DESCRIPTION
**Pull request by amyspark**
- [X] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [X] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [X] Have you successfully run `brew style` with your changes locally?
- [X] Have you successfully run `brew tests` with your changes locally?

-----

Fixes Homebrew/brew#6386.

### CODE REVIEW

File: Library/Homebrew/cmd/upgrade.rb [hunk 1]
```diff
@@ -213,37 +213,42 @@ def upgradable_dependents(kegs, formulae)
     formulae_to_upgrade = Set.new
     formulae_pinned = Set.new
 
-    formulae.each do |formula|
+    formulae_to_check = formulae
+    checked_formulae = Set.new
+
+    until formulae_to_check.empty?
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cmd/upgrade.rb]: I worry about this still creating an infinite loop. Is this avoidable? Perhaps a max depth could be configured?

**Comment by MikeMcQuaid** [comment]
Run and commit `brew style --fix` to fix CI.

**Comment by MikeMcQuaid** [comment]
Otherwise: looking good, thanks!

File: Library/Homebrew/cmd/upgrade.rb (hunk 1 above)
Comment (reitermarkus) [reply Library/Homebrew/cmd/upgrade.rb]: I think that's what the 

```
next if checked_formulae.include?(formula)
```

below is for.

Comment (amyspark) [reply Library/Homebrew/cmd/upgrade.rb]: @reitermarkus is correct, the trick lies in that line.

**Comment by MikeMcQuaid** [comment]
Thanks again @amyspark!

**Pull request merged**